### PR TITLE
Try avoiding the deprecated findDOMNode API from DropZone Provider

### PR DESCRIPTION
**Pull request by youknowriad**
While I don't like adding DOM nodes for nothing (we can probably avoid these nodes if we use hooks), I don't see another way to remove `findDOMNode` and keep the same functionality.

Thoughs @aduth @gziolo

### CODE REVIEW

File: packages/components/src/drop-zone/provider.js [hunk 1]
```diff
@@ -212,7 +209,7 @@ class DropZoneProvider extends Component {
 		const { position, hoveredDropZone } = this.state;
 		const dragEventType = getDragEventType( event );
 		const dropZone = this.dropZones[ hoveredDropZone ];
-		const isValidDropzone = !! dropZone && this.container.contains( event.target );
+		const isValidDropzone = !! dropZone && this.container.current.contains( event.target );
```
Comment (aduth) [inline packages/components/src/drop-zone/provider.js]: If this is the only instance of the DOM reference, could we just have `onDrop` be bound as an event handler to the wrapping `div`? I guess the main difference is the lack of `stopPropagation` / `preventDefault` on all other drops on the page, which I could see as equally "correct" and "annoying" (in cases of mis-drops causing navigation).

Comment (aduth) [reply packages/components/src/drop-zone/provider.js]: Also, with use of Slot/Fill, using DOM `Element#contains` isn't guaranteed to be accurate anyways.

Comment (aduth) [reply packages/components/src/drop-zone/provider.js]: If this provider is at the top-level, is there really any point to checking this at all?

Assuming we're tracking hover for the relevant `dropZone`, that seems to be an extra (better?) level of certainty to where the drop is taking place.

**Review by aduth — CHANGES_REQUESTED** [review]
This breaks "click below to add" behavior, because of the added `div` (the WritingFlow relies on `100%` height of its full ancestry)

**Comment by aduth** [comment]
There was meant to be an E2E test covering the broken behavior. Apparently it's not working so well.

Originally: https://github.com/WordPress/gutenberg/pull/5991/files#diff-708f5078fca40c1d3de2c973ddc5e255

Current:

https://github.com/WordPress/gutenberg/blob/a2f81faa58afbbcb28d68ef879a5354c257b2d85/test/e2e/specs/adding-blocks.test.js#L17-L18

File: packages/components/src/drop-zone/provider.js (hunk 1 above)
Comment (youknowriad) [reply packages/components/src/drop-zone/provider.js]: Yes, there is, this fixes a bug where drag and dropping to the media library creates the image twice. (It's also caught by the dropzones in the editor even if the the dragged item is outside the editor)

Comment (aduth) [reply packages/components/src/drop-zone/provider.js]: Do you recall the issue / pull request for it? This sounds a bit fishy...

Makes me wonder if there's possibly something going on with React DevTools double-rendering.

Comment (youknowriad) [reply packages/components/src/drop-zone/provider.js]: So the original PR fixing this issue is this one https://github.com/WordPress/gutenberg/pull/5432

but I recall it being refactored to use the container technique in another PR. Looking

Comment (youknowriad) [reply packages/components/src/drop-zone/provider.js]: Looks like it was introduced here https://github.com/WordPress/gutenberg/pull/4115 but I don't recall the exact reasons :( Will investigate more.

Comment (youknowriad) [reply packages/components/src/drop-zone/provider.js]: So yes, in my testing, it is still necessary to avoid the double upload issue.

**Comment by youknowriad** [comment]
I updated the PR. I fixed the e2e test to break properly if needed. And I also fixed the issue using CSS, I don't have a better alternative at the moment.

**Comment by aduth** [comment]
Aside: I really wish `display:contents` was better supported. It would be perfect for these sorts of wrappers.

**Comment by aduth** [comment]
Maybe I'm missing something? I don't see the issue resolved by #5432 with c869c81.

**Comment by youknowriad** [comment]
@aduth I suspect it's because the drop event is triggered in that case because the div container is rendered behind the modal. I'm checking the `event.target` which is a bit different.

**Comment by youknowriad** [comment]
Or are you saying it's fixed? Because I can't reproduce in my testing, your commit seems fine.

**Comment by aduth** [comment]
Right, c869c81 is what I was trying to suggest, and appears to work well enough. Will approve if it's agreeable to you.

**Review by aduth — APPROVED** [review]

**Pull request merged**